### DISCUSSION TITLE
Using ruff as a Rust library

### DISCUSSION THREAD

**Discussion by realitycheck** [astral-sh/ruff#8755] [Q&A]
AFAIK, there are a lot of `pub(crate)` statements in ruff_cli crate instead of just `pub` which definitely preventing to use ruff as a Rust library. 

Is this intentional design approach? What's the authors' concern here?

**Comment by zanieb** [comment]
Hi! Ruff is not intended to be used from Rust yet i.e. we do not have a public Rust API and we are not publishing to crates.io. There's some intention to allow programmatic access to Ruff outside the CLI in the future, but we are not prioritizing that work right now. In large part, the concern is that we are not ready to stabilize these APIs and would rather focus on improving the current public user interface.
